src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -179,23 +179,20 @@ public CSVFileScanFragment(
   @Override
   protected TupleBatch fetchNextReady() throws IOException, DbException {
     long lineNumberBegin = lineNumber;
-    boolean unfinishedEncapsulatedTokenRow = false;
+    boolean truncatedQuotedField = false;
 
     while ((buffer.numTuples() < TupleBatch.BATCH_SIZE) && !flagAsIncomplete) {
       lineNumber++;
       if (parser.isClosed()) {
         break;
       }
 
-      if (unfinishedEncapsulatedTokenRow) {
+      if (truncatedQuotedField) {
         onLastRow = true;
       }
 
-      try {
-        if (!onLastRow) {
-          record = iterator.next();
-        }
-      } catch (Exception e) {
+      if (!onLastRow) {
+        record = iterator.next();
```

Comment:
Also, we can't flip the order between ```next()``` and ```hasNext()``` (based on your earlier suggestion) because this results on the last row being read twice (once we hit the last row, we are not flagging it as last). It seems like ```hasNext()``` has to come before ```next()```.